File: src/main/scala/workshop/part2/ComputeActorFactory.scala
package workshop.part2

import scala.concurrent.duration._
import akka.actor.{Props, ActorRef, ActorContext}
import workshop.part1.ComputeActor


class ComputeActorFactory(logCompletedTasksInterval: FiniteDuration = 1 second) {
  def create(context: ActorContext, actorName: String): ActorRef = {
    context.actorOf(Props(classOf[ComputeActor], logCompletedTasksInterval), actorName)
  }
}

File: src/main/java/workshop/part2/ComputeActorFactory.java
package workshop.part2;

import akka.actor.ActorContext;
import akka.actor.ActorRef;
import akka.actor.Props;
import scala.concurrent.duration.Duration;
import scala.concurrent.duration.FiniteDuration;
import workshop.part1.ComputeActor;

import java.util.concurrent.TimeUnit;

public class ComputeActorFactory {

    static FiniteDuration logCompletedTasksInterval = Duration.create(1, TimeUnit.SECONDS);
    private ActorRef numCompletedTaskActor;

    public ComputeActorFactory(ActorRef numCompletedTaskActor) {
        this.numCompletedTaskActor = numCompletedTaskActor;
    }

    public ActorRef create(ActorContext context, String actorName) {
        return context.actorOf(Props.create((ComputeActor.class), numCompletedTaskActor, logCompletedTasksInterval), actorName);
    }
}